lib/AST/Module.cpp

```diff
@@ -1024,6 +1049,53 @@ void ModuleDecl::getTopLevelDeclsWhereAttributesMatch(
   FORWARD(getTopLevelDeclsWhereAttributesMatch, (Results, matchAttributes));
 }
 
+void ModuleDecl::lookupTopLevelDeclsByObjCName(SmallVectorImpl<Decl *> &Results,
+                                               DeclName name) {
+  if (!isObjCNameLookupCachePopulated())
+    populateObjCNameLookupCache();
+
+  if (name.isSpecial())
```

Comment:
Are we dropping anything here? I modeled this after the SwiftLookupTable, which also seems to bail on "special" names: https://github.com/apple/swift/blob/main/lib/ClangImporter/SwiftLookupTable.h#L69-L71.

Reply:
I suppose if you wrote `@class init`, `@class dealloc` or `@class subscript`, maybe? Does merely use the same spelling make them special? I somehow doubt it, but I will check.

Reply:
Added it to a test, seems fine.